ui-library-filesharing-ui-components/api/UploadFileToAzureBlobStore/index.ts

```diff
@@ -1,53 +1,62 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 import { AzureFunction, Context, HttpRequest } from "@azure/functions";
-import * as multipart from "parse-multipart";
+import {
+  BlobSASPermissions,
+  BlobServiceClient,
+  SASProtocol,
+} from "@azure/storage-blob";
 import HTTP_CODES from "http-status-enum";
+import * as multipart from "parse-multipart";
 
 const azureStorageConnectionString =
   process.env["azureStorageConnectionString"];
-let storageAccountName;
+
+let storageAccountName: string | undefined;
+
 try {
   storageAccountName =
     azureStorageConnectionString.match(/AccountName=([^;]+)/)[1];
 } catch {
-  console.error("Invalid azure storage connection string.");
+  throw new Error("Invalid Azure Storage Connection String");
 }
 
-// Azure storage container name to be used for uploading files.
-// Must match the value in function.json `path`
+// Name of the container in Azure Storage to store all files in.
+// Make sure that this name matches the value in the 'function.json' file.
 const fileSharingUploadsContainerName = "uploads";
 
 const httpTrigger: AzureFunction = async function (
   context: Context,
   req: HttpRequest
-): Promise<void> {
-  context.log(
-    "HTTP trigger function processed a request to upload a file to azure blob store."
-  );
+): Promise<unknown> {
+  context.log("UploadFileToAzureBlobStore: Processing new request.");
 
   if (!req.body || !req.body.length) {
     context.res.body = { error: `Request body is not defined` };
     context.res.status = HTTP_CODES.BAD_REQUEST;
     return;
   }
 
-  if (!req.query?.filename) {
+  const filename = req.query?.filename;
+
+  // `filename` is required property to correctly upload the file in the provided storage
+  if (!filename) {
     context.res.body = { error: `filename is not defined` };
     context.res.status = HTTP_CODES.BAD_REQUEST;
     return;
   }
 
+  // Content type is required to know how to parse multi-part form
   if (!req.headers || !req.headers["content-type"]) {
     context.res.body = {
-      error: `Content type is not set in header 'content-type'`,
+      error: `Content type is not sent in header 'content-type'`,

```

Comment:
This should be set; the error is that the content-type header is not set on the request

Reply:
Copied the code from filesharing-chat-composite, will correct it there also